From 4f59b0a437416e001ba96be22dfcd39806af70e6 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Fri, 6 Nov 2020 18:50:33 +0000
Subject: [PATCH] build: Enable MemorySSA DSE with pinned clang.

The feature has been on on the tot bots for a while now and it's been
on by default upstream for a while, and things seem to work.

Since we just rolled in a clang with all the fixes, let's turn this on.

Bug: 1127713
Change-Id: I56805d576bea5936d58c653da994dc738c2a79a3
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2520013
Commit-Queue: Nico Weber <thakis@chromium.org>
Reviewed-by: Hans Wennborg <hans@chromium.org>
Cr-Commit-Position: refs/heads/master@{#824947}
---
 build/config/compiler/BUILD.gn | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 6f8c779040251..52912b1cb546a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -490,25 +490,6 @@ config("compiler") {
       "-mllvm",
       "-instcombine-lower-dbg-declare=0",
     ]
-
-    # Pinned clang has enable-dse-memoryssa enabled by default but it's broken,
-    # so we have to force it off.
-    # Trunk clang has it disabled by default but it might work, so we force it
-    # on so the ToT bots can check if it works now.
-    if (!llvm_force_head_revision) {
-      cflags += [
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm",
-        "-enable-dse-memoryssa=false",
-      ]
-    }
-    if (llvm_force_head_revision) {
-      cflags += [
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm",
-        "-enable-dse-memoryssa=true",
-      ]
-    }
   }
 
   # C11/C++11 compiler flags setup.
